src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -482,27 +482,25 @@ private[internal] trait GlbLubs {
       try {
         val (ts, tparams) = stripExistentialsAndTypeVars(ts0)
         val glbOwner = commonOwner(ts)
-        def refinedToParents(t: Type): List[Type] = t match {
-          case RefinedType(ps, _) => ps flatMap refinedToParents
-          case _ => List(t)
-        }
-        def refinedToDecls(t: Type): List[Scope] = t match {
-          case RefinedType(ps, decls) =>
-            val dss = ps flatMap refinedToDecls
-            if (decls.isEmpty) dss else decls :: dss
-          case _ => List()
+        val ts1 = {
+          val res = mutable.ListBuffer.empty[Type]
```

Comment:
If we could add the elements here in reverse order, we could halve the allocations at this point.